app/components/FloatingAction.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react"
+import { Dimensions, View, ViewStyle } from "react-native"
+import Animated, { FadeInDown, FadeOutDown } from "react-native-reanimated"
+import { spacing } from "../theme"
+
+type Props = {
+  children: React.ReactElement
+  isScrolling?: boolean
+}
+
+enum Scroll {
+  onMomentumScrollBegin = "onMomentumScrollBegin",
+  onMomentumScrollEnd = "onMomentumScrollEnd",
+  onScrollBeginDrag = "onScrollBeginDrag",
+  onScrollEndDrag = "onScrollEndDrag",
+}
+
+export const useFloatingActionEvents = () => {
```

Comment:
nit: this is a great hook, but isn't really related to the Floating button. Could we move this out into `app/hooks` as `useScrollHandlers`, or something similar?